src/ublox/ubxmsg.h

```diff
@@ -474,19 +466,19 @@ namespace ublox {
         } __attribute__((packed))
         flags;
 
-        static gps_fix::status_t to_status( enum status_t status, flags_t flags )
+        static gps_fix::status_t to_status( enum nav_status_t::status_t status, flags_t flags )
         {
             if (!flags.gps_fix)
-            return gps_fix::STATUS_NONE;
+              return gps_fix::STATUS_NONE;
             if (flags.diff_soln)
-            return gps_fix::STATUS_DGPS;
+              return gps_fix::STATUS_DGPS;
             switch (status) {
-                case NAV_STAT_DR_ONLY  : return gps_fix::STATUS_EST;
-                case NAV_STAT_2D       :
-                case NAV_STAT_3D       :
-                case NAV_STAT_GPS_DR   : return gps_fix::STATUS_STD;
-                case NAV_STAT_TIME_ONLY: return gps_fix::STATUS_TIME_ONLY;
-                default                : return gps_fix::STATUS_NONE;
+              case nav_status_t::status_t::NAV_STAT_DR_ONLY  : return gps_fix::STATUS_EST;
```

Comment:
(continued from line 993)

Since the `status` value is actually a `gps_fix::status_t`, I think you can just return the first argument instead of re-mapping with another `switch` statement:

```
         return status;
```
... assuming the first argument type is changed to `gps_fix::status_t`.  I don't think the `m_fix.status` is set correctly when it is mapped twice.